management_api.go

```diff
@@ -411,10 +411,38 @@ func (e *Enforcer) UpdateNamedGroupingPolicies(ptype string, oldRules [][]string
 
 // RemoveFilteredNamedGroupingPolicy removes a role inheritance rule from the current named policy, field filters can be specified.
 func (e *Enforcer) RemoveFilteredNamedGroupingPolicy(ptype string, fieldIndex int, fieldValues ...string) (bool, error) {
-	return e.removeFilteredPolicy("g", ptype, fieldIndex, fieldValues...)
+	return e.removeFilteredPolicy("g", ptype, fieldIndex, fieldValues)
 }
 
 // AddFunction adds a customized function.
 func (e *Enforcer) AddFunction(name string, function govaluate.ExpressionFunction) {
 	e.fm.AddFunction(name, function)
 }
+
+func (e *Enforcer) SelfAddPolicy(sec string, ptype string, rule []string) (bool, error) {
```

Comment:
My opinion is to extract the code in ``addPolicy`` and put it into ``SelfAddPolicy``, and then call ``SelfAddPolicy`` in ``addPolicy`` instead of modifying the original parameters of ``addPolicy``.
like this
```
func (e *Enforcer) addPolicy(sec string, ptype string, rule []string) (bool, error) {
	e.SelfAddPolicy(sec,ptype,rule)
	if e.watcher != nil && e.autoNotifyWatcher {

        }
```